src/main/java/io/github/railroad/vcs/connections/hubs/GithubConnection.java

```diff
@@ -0,0 +1,218 @@
+package io.github.railroad.vcs.connections.hubs;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import io.github.railroad.Railroad;
+import io.github.railroad.vcs.Repository;
+import io.github.railroad.vcs.RepositoryTypes;
+import io.github.railroad.vcs.connections.AbstractConnection;
+import io.github.railroad.vcs.connections.Profile;
+import javafx.scene.image.Image;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class GithubConnection extends AbstractConnection {
+    private Profile account;
+
+    public GithubConnection(Profile profile) {
+        this.account = profile;
+    }
+
+    private List<HttpResponse> readHTTP(String method, String postixurl, String body) throws Exception {
+        return readHTTP(method, postixurl, body, true);
+    }
+
+    private List<HttpResponse> readHTTP(String method, String postixurl, String body, boolean enable_pages) throws Exception {
+        if (account.getAccessToken().isEmpty()) {
+            throw new Exception("Missing Access Token");
+        }
+        List<HttpResponse> result = new ArrayList<>();
+
+        String request_url = "https://api.github.com/" + postixurl;
+        boolean finished = false;
+        try {
+            try {
+
```

Comment:
I don't see the need for the nested try-catch here.

Reply:
removed as your comment is true